lxd/storage/dir_utils.go

```diff
@@ -0,0 +1,39 @@
+package storage
+
+import (
+	"os"
+
+	"github.com/lxc/lxd/lxd/project"
+	"github.com/lxc/lxd/shared"
+)
+
+// DirSnapshotDeleteInternal deletes the given snapshot internally.
+func DirSnapshotDeleteInternal(projectName, poolName string, snapshotName string) error {
```

Comment:
@monstermunchkin can this function be made unexported from the `storage` package if it is internal?

Reply:
Interestingly, this function is not actually an internal function. It's used in `lxd/api_internal.go`. It therefore needs to be exported. Perhaps we should think of renaming this function :)